src/otx/algo/hooks/recording_forward_hook.py

```diff
@@ -364,3 +366,56 @@ def func(
             saliency_maps = self._normalize_map(saliency_maps)
 
         return saliency_maps.reshape((batch_size, self._num_classes, height, width))
+
+
+class MaskRCNNRecordingForwardHook(BaseRecordingForwardHook):
+    """Dummy saliency map hook for Mask R-CNN model."""
+
+    @classmethod
+    def create_and_register_hook(cls) -> BaseRecordingForwardHook:
+        """Create this object and register it to the module forward hook."""
+        return cls()
+
+    def func(self, feature_map: torch.Tensor | Sequence[torch.Tensor], _: int = -1) -> None:
+        """Return None for saliency map hook to generate saliency maps from predictions."""
+        return
+
+    @classmethod
+    def get_sal_map_from_preds(cls, preds: list[InstanceSegBatchPredEntity], num_classes: int) -> torch.Tensor:
+        """Generate saliency maps from predicted masks by averaging and normalizing them per-class.
+
+        Args:
+            preds (List[InstanceSegBatchPredEntity]): Predictions of Instance Segmentation model.
+            num_classes (int): Num classes that model can predict.
+
+        Returns:
+            torch.Tensor: Class-wise Saliency Maps. One saliency map per each class - [batch, class_id, H, W]
+        """
+        # TODO(gzalessk): Add unit tests # noqa: TD003
+        batch_size = len(preds)
+        _, height, width = preds[0].masks[0].data.shape
+        batch_saliency_maps = torch.empty((batch_size, num_classes, height, width))
```

Comment:
There is no guarantee that `preds.masks` has the same dimensions among themselves. How about revising this this function to take `InstanceSegBatchPredEntity` instead?

Reply:
Great point, thank you